test/cppsuite/test_harness/workload_generator.h

```diff
@@ -173,48 +179,88 @@ class workload_generator : public component {
             break;
         case thread_operation::REMOVE:
         case thread_operation::INSERT:
-        case thread_operation::UPDATE:
             /* Sleep until it is implemented. */
             while (context.is_running())
                 std::this_thread::sleep_for(std::chrono::seconds(1));
             break;
+        case thread_operation::UPDATE:
+            update_operation(context, session);
+            break;
         default:
             testutil_die(DEBUG_ABORT, "system: thread_operation is unknown : %d",
               static_cast<int>(context.get_thread_operation()));
             break;
         }
     }
 
-    /* Basic read operation that walks a cursors across all collections. */
+    /* 
+     * Basic update operation that currently update the same key with a random value in each
+     * collection.
+     */
     static void
-    read_operation(thread_context &context, WT_SESSION *session)
+    update_operation(thread_context &context, WT_SESSION *session)
     {
         WT_CURSOR *cursor;
         WT_DECL_RET;
-        std::string config;
+        wt_timestamp_t ts;
         std::vector<WT_CURSOR *> cursors;
+        std::vector<std::string> collection_names;
+        std::string generated_value;
         bool has_committed = true;
+        int cpt;
 
         testutil_assert(session != nullptr);
         /* Get a cursor for each collection in collection_names. */
         for (const auto &it : context.get_collection_names()) {
             testutil_check(session->open_cursor(session, it.c_str(), NULL, NULL, &cursor));
             cursors.push_back(cursor);
+            collection_names.push_back(it);
         }
 
         while (context.is_running()) {
             /* Walk each cursor. */
-            context.begin_transaction(session, config);
+            context.begin_transaction(session, "");
+            ts = context.set_commit_timestamp(session);
+            cpt = 0;
             for (const auto &it : cursors) {
-                if ((ret = it->next(it)) != 0)
-                    it->reset(it);
+                int value_size = 10;
```

Comment:
Currently hard coded as we do not have access to the configuration parameter from here, what do you suggest ?

Reply:
We could add a value_size member to the random_generator, or add the config to the thread context or create a new function with a random value size but then the configuration parameter will lose its interest.

Reply:
As discussed, value_size will be given to the thread_context as a temp solution.